libobs/obs-hotkey.c

```diff
@@ -1328,6 +1330,30 @@ void obs_hotkey_enable_strict_modifiers(bool enable)
 	unlock();
 }
 
+void obs_hotkey_external_set_pressed(obs_key_t key, bool pressed)
+{
+	if (key == OBS_KEY_NONE)
+		return;
+
+	if (pressed)
```

Comment:
To ensure thread safety (between the hotkey thread and the UI thread/Qt), I'd suggest wrapping write access in locks, like so:

```
    if (!lock())
        return;

    if (pressed)
		da_push_back(obs->hotkeys.external_keys, &key);
	else
		da_erase_item(obs->hotkeys.external_keys, &key);
    unlock();
```